docs/book/src/development/development.md

```diff
@@ -342,6 +342,76 @@ $ sudo ln -s /run/user/$(id -u)/podman/podman.sock /var/run/docker.sock
 [hack-ci-gce-project]: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/hack/ci/gce-project.sh
 [hack-ci-openstack]: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/hack/ci/openstack.sh
 
+## Cluster Creation with ClusterClass and Tilt
+
+This section provides a guide to create a Kubernetes cluster using ClusterClass in the development environment. It covers the process of setting up environment variables, creating a Kind cluster, configuring a secret, applying ClusterClass, deploying the cluster, monitoring its provisioning, obtaining Kubeconfig, and applying CNI to the cluster.
+
+#### Setting Environment Variables
+
+Define necessary environment variables relevant to your cluster and infrastructure. This includes setting variables for SSH key names and Kubernetes version.
+
+```bash
+#### Set the name for the SSH keypair used for instances in OpenStack
+export OPENSTACK_SSH_KEY_NAME="your_ssh_key_name_here"
+#### Set the desired Kubernetes version for your cluster
+export KUBERNETES_VERSION="v1.28.5"
+```
+
+#### Creating a Kind Cluster
+
+Create a Kind cluster. This process involves the initialization of Cluster API providers for OpenStack.
+
+```bash
+kind create cluster
+export CLUSTER_TOPOLOGY=true
+clusterctl init --infrastructure openstack
+```
+
+#### Secret Configuration
+
+Apply a Kubernetes secret that your cluster configuration will reference. Ensure the secret's name aligns with the required format in your configuration.
+
+#### Applying ClusterClass and Cluster Configuration
+
+Substitute variables in your ClusterClass configuration file and apply it to set up the cluster class.
+
+```bash
+envsubst < templates/clusterclass-dev-test.yaml | kubectl apply -f -
+envsubst < templates/cluster-template-development.yaml | kubectl apply -f -
+```
+
+#### Setting up CNI and External Cloud Provider
+
+Create a configuration file `cloud.conf` for the external cloud provider. This file should contain necessary details for cloud provider integration:
+
+```bash
+[Global]
+auth-url=<your_auth_url>
+application-credential-id=<your_credential_id>
+application-credential-secret=<your_credential_secret>
+region=<your_region>
+domain-name=<your_domain_name>
+```
+
+After setting up the cloud.conf file, apply this configuration and the Container Network Interface (CNI) to your workload cluster:
+
+```bash
+clusterctl get kubeconfig $CLUSTER_NAME > kubeconfig.yaml
+kubectl --kubeconfig=kubeconfig.yaml apply -k <path_to_your_cni_configuration>
+```
+
+#### Using ClusterClass with Tilt for Development
+
+To utilize Tilt for an efficient and streamlined development process, add the following section under `kustomize_substitutions:` in your `tilt-settings.yaml` file:
+
+```yaml
+kustomize_substitutions:
+  CLUSTER_TOPOLOGY: "true"
+  OPENSTACK_SSH_KEY_NAME: "your_ssh_key_name_here"
+  KUBERNETES_VERSION: "v1.28.5"
+  NAMESPACE: "default"
```

Comment:
If you use ClusterClass variables to set the version this will no longer be necessary. Then we can set the version directly from the Tilt UI. The namespace can already be set from there.
```suggestion
```